php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -841,20 +845,28 @@ private function convertJsonValueToProtoValue(
                         "Bool field only accept bool value");
                 }
                 return $value;
-            case GPBType::FLOAT:
+            case GPBType::ENUM:
                 if (is_null($value)) {
                     return $this->defaultValue($field);
                 }
-                if ($value === "Infinity") {
-                    return INF;
+                if (is_integer($value)) {
+                    return $value;
                 }
-                if ($value === "-Infinity") {
-                    return -INF;
+                $enum_value = $field->getEnumType()->getValueByName($value);
+                if (!is_null($enum_value)) {
+                    return $enum_value->getNumber();
                 }
-                if ($value === "NaN") {
-                    return NAN;
+                throw new GPBDecodeException(
+                        "Enum field only accepts bool value");
+            case GPBType::STRING:
```

Comment:
Could you move ENUM and STRING cases to their original places?

Reply:
yep, good call